### PR TITLE
[Fluid] Div-stable P2P1 element

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
This PR adds a P2P1 element (together with its corresponding condition). The automatic differentiation files for generating the element are added as well. Regarding the condition, I should note that the drag calculation, the pure slip correction contribution and the wall laws support are left for a future implementation as these require a bit more intricate developments for which we don't have resources to put on at the moment.

Besides, an auxiliary function to postprocess the edges' midpoint nodes pressure has been also added (note that this is called at the end of the `SolveSolutionStep` considering that current element can be used in an outer nonlinear loop).  This function works with both OpenMP and MPI parallelism.

Subsequent PRs will make this element available in the GiD and FlowGraph GUIs and add a use case to the examples repository.

### CODE REVIEW

**Comment by ddiezrod** [comment]
Just out of curiosity: I thought that the advantage of this element is that you do not need to add all the stabilization terms, but it seems that is not the case here, is there any reason for this? did you try it without any pressure stabilization?

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla Just out of curiosity: I thought that the advantage of this element is that you do not need to add all the stabilization terms, but it seems that is not the case here, is there any reason for this? did you try it without any pressure stabilization?

You're right, pressure stabilization is not strictly needed in here as this interpolation pair is LBB compliant. Indeed, I tried both with and without and works in both cases. However, it is known to behave better if the pressure subscale is added (that's why I chose this as default option).

**Comment by rubenzorrilla** [comment]
Any good samaritan willing to have a look to this?

File: applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp [hunk 1]
```diff
@@ -41,8 +41,8 @@ void NavierStokesWallCondition<TDim,TNumNodes,TWallModel...>::EquationIdVector(
 
     IndexType local_index = 0;
     const auto& r_geometry = GetGeometry();
-    const unsigned int v_pos = r_geometry[0].GetDofPosition(DENSITY);
-    const unsigned int p_pos = r_geometry[0].GetDofPosition(MOMENTUM);
+    const unsigned int v_pos = r_geometry[0].GetDofPosition(VELOCITY_X);
+    const unsigned int p_pos = r_geometry[0].GetDofPosition(PRESSURE);
```
Comment (ddiezrod) [inline applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp]: Shouldn't this belong to another PR?

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp]: It was already merged in #12335 (I think I should merge the master in here).

**Comment by jginternational** [comment]
Remember to merge https://github.com/KratosMultiphysics/GiDInterface/pull/971 after

**Comment by rubenzorrilla** [comment]
> Remember to merge [KratosMultiphysics/GiDInterface#971](https://github.com/KratosMultiphysics/GiDInterface/pull/971) after

Sure! Thanks for the reminder in any case.

**Review by RiccardoRossi — APPROVED** [review]
it looks in good shape to me
